src/layers/LevelLayer/LevelLayer.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-param-reassign */
+import MapboxStyleLayer from '../MapboxStyleLayer';
+
+/**
+ * Layer for visualizing station levels.
+ *
+ * @class
+ * @param {Object} [options] Layer options.
+ * @inheritdoc
+ */
+class LevelLayer extends MapboxStyleLayer {
+  /**
+   * Apply visibility to style layers that fits the filter function.
+   */
+  // eslint-disable-next-line class-methods-use-this
+  applyLayoutVisibility(mbMap, visible, filterFunc, layerFilter, isRadio) {
+    const style = mbMap.getStyle();
+
+    if (!mbMap || !style || (isRadio && !visible)) {
```

Comment:
is isRadio necessary?

Reply:
It is the solution I found to only instantiate the visible layer of the radio.
To avoid that if the last layer is not visible, it will toggle off all the filters

Reply:
I onlt set style for the visible LevelLayer now